model/src/main/kotlin/util/FindingCurationMatcher.kt

```diff
@@ -0,0 +1,80 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model.util
+
+import com.here.ort.model.LicenseFinding
+import com.here.ort.model.config.LicenseFindingCuration
+import com.here.ort.spdx.SpdxLicense
+
+import java.nio.file.FileSystems
+import java.nio.file.Paths
+
+/**
+ * A class for matching and applying [LicenseFindingCuration]s to [LicenseFinding]s.
+ */
+class FindingCurationMatcher {
+    private fun isPathMatching(finding: LicenseFinding, curation: LicenseFindingCuration): Boolean =
+        FileSystems.getDefault().getPathMatcher("glob:${curation.path}").let {
+            it.matches(Paths.get(finding.location.path))
+        }
+
+    private fun isStartLineMatching(finding: LicenseFinding, curation: LicenseFindingCuration): Boolean =
+        curation.startLines.isEmpty() || curation.startLines.any { it.equals(finding.location.startLine) }
+
+    private fun isLineCountMatching(finding: LicenseFinding, curation: LicenseFindingCuration): Boolean =
+        curation.lineCount == null || curation.lineCount == finding.location.endLine - finding.location.startLine + 1
+
+    private fun isDetectedLicenseMatching(finding: LicenseFinding, curation: LicenseFindingCuration): Boolean =
+        curation.detectedLicense == finding.license
+
+    /**
+     * Return true if and only if the given curation is applicable to the given finding.
+     */
+    fun matches(finding: LicenseFinding, curation: LicenseFindingCuration): Boolean =
+        isPathMatching(finding, curation) &&
+                isStartLineMatching(finding, curation) &&
+                isLineCountMatching(finding, curation) &&
+                isDetectedLicenseMatching(finding, curation)
+
+    /**
+     * Return the curated finding if the given [curation] is applicable to the given [finding] or the given [finding]
+     * otherwise. Null is returned if and only if the given curation is applicable and its concluded license equals
+     * [SpdxLicense.NONE].
+     */
+    fun apply(finding: LicenseFinding, curation: LicenseFindingCuration): LicenseFinding? =
+        if (!matches(finding, curation)) finding
+        else if (curation.concludedLicense == SpdxLicense.NONE) null
+        else finding.copy(license = curation.concludedLicense)
+
+    /**
+     * Applies the given [curations] to the given [findings]. In case multiple curations apply to any given finding only
+     * an arbitrary single one is applied.
+     */
+    fun applyAll(
+        findings: Collection<LicenseFinding>,
+        curations: Collection<LicenseFindingCuration>
+    ): List<LicenseFinding> =
+        findings.mapNotNull { finding ->
+            curations.find { matches(finding, it) }.let {
+                if (it != null) apply(finding, it)
+                else finding
+            }
+        }
```

Comment:
The apply functions do not fit well into the matcher class. Why not put them in to the `LicenseFindingCuration` or the `Curations` class?

Reply:
I think it does fit quite well as the purpose of this class is ```applying curations```. Would the issue you mention be fixed by renaming the class ? I don't have a nice idea though, how about ```LicenseFindingCurator```?

I didn't put it into the ```Curation``` class because:
1. The responsibility of applying curations seemed like complex enough to put it into a separate class, e.g. it seemed to be one reposibility which can be considered in isolation. 
2. Coupling
  - no dependencies between ```LicenseFindingCuration```and ```LicenseFinding```
  - no dependencies from ```LicenseFindingCuration``` to the matching logic

Reply:
Ok, let's merge it like that, I can propose a refactoring if I feel it's required.